NDTensors/src/lib/Sectors/src/category_product.jl

```diff
@@ -34,78 +45,206 @@ function Base.show(io::IO, s::CategoryProduct)
 
-#
-# Dictionary-like implementation
-#
+function ×(l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger)
+  c3 = LabelledNumbers.label(l1) × LabelledNumbers.label(l2)
+  m3 = LabelledNumbers.unlabel(l1) * LabelledNumbers.unlabel(l2)
+  return LabelledNumbers.labelled(m3, c3)
+end
+
+function ×(g1::AbstractUnitRange, g2::AbstractUnitRange)
+  v = map(
+    ((l1, l2),) -> l1 × l2,
+    Iterators.flatten((
+      Iterators.product(BlockArrays.blocklengths(g1), BlockArrays.blocklengths(g2)),
+    ),),
+  )
+  return GradedAxes.gradedrange(v)
+end
+
+# ====================================  Fusion rules  ======================================
+# generic case: fusion returns a GradedAxes, even for fusion with Empty
+function fusion_rule(::SymmetryStyle, s1::CategoryProduct, s2::CategoryProduct)
+  return to_graded_axis(categories_fusion_rule(categories(s1), categories(s2)))
+end
+
+# Abelian case: fusion returns CategoryProduct
+function fusion_rule(::AbelianGroup, s1::CategoryProduct, s2::CategoryProduct)
+  return categories_fusion_rule(categories(s1), categories(s2))
+end
+
+# Empty case
+function fusion_rule(
+  ::EmptyCategory, ::CategoryProduct{Tuple{}}, ::CategoryProduct{Tuple{}}
+)
+  return sector()
+end
+
+# EmptyCategory acts as trivial on any AbstractCategory, not just CategoryProduct
+function fusion_rule(::SymmetryStyle, ::CategoryProduct{Tuple{}}, c::AbstractCategory)
+  return to_graded_axis(c)
+end
+function fusion_rule(::SymmetryStyle, ::CategoryProduct{Tuple{}}, c::CategoryProduct)
+  return to_graded_axis(c)
+end
+function fusion_rule(::SymmetryStyle, c::AbstractCategory, ::CategoryProduct{Tuple{}})
+  return to_graded_axis(c)
+end
+function fusion_rule(::SymmetryStyle, c::CategoryProduct, ::CategoryProduct{Tuple{}})
+  return to_graded_axis(c)
+end
+
+# abelian case: return Category
+fusion_rule(::AbelianGroup, ::CategoryProduct{Tuple{}}, c::AbstractCategory) = c
+fusion_rule(::AbelianGroup, ::CategoryProduct{Tuple{}}, c::CategoryProduct) = c
+fusion_rule(::AbelianGroup, c::AbstractCategory, ::CategoryProduct{Tuple{}}) = c
+fusion_rule(::AbelianGroup, c::CategoryProduct, ::CategoryProduct{Tuple{}}) = c
+
+# ===============================  Ordered implementation  =================================
+CategoryProduct(t::Tuple) = _CategoryProduct(t)
+CategoryProduct(cats::AbstractCategory...) = CategoryProduct((cats...,))
+
+categories_isequal(o1::Tuple, o2::Tuple) = (o1 == o2)
+
+categories_isless(::Tuple, ::Tuple) = throw(ArgumentError("Not implemented"))
+categories_isless(t1::T, t2::T) where {T<:Tuple} = t1 < t2
 
+categories_product(l1::Tuple, l2::Tuple) = (l1..., l2...)
+
+categories_trivial(type::Type{<:Tuple}) = trivial.(fieldtypes(type))
+
+function find_common(t1::Tuple, t2::Tuple)
+  n = min(length(t1), length(t2))
+  return t1[begin:n], t2[begin:n]
+end
+
+function find_diff(t1::Tuple, t2::Tuple)
+  n1 = length(t1)
+  n2 = length(t2)
+  return n1 < n2 ? t2[(n1 + 1):end] : t1[(n2 + 1):end]
+end
```

Comment:
In Julia, functions that start with `find` by convention output an index, not a value, for example see [findfirst](https://docs.julialang.org/en/v1/base/arrays/#Base.findfirst-Tuple{Any}).

Maybe better names would be `categories_common` and `categories_diff`.